ddtrace/debugging/_capture/snapshot.py

```diff
@@ -53,68 +60,104 @@ class Snapshot(CapturedEvent):
+
     def enter(self):
         frame = self.frame
-        probe = cast(SnapshotFunctionProbe, self.probe)
-        _args = list(self.args or safe_getter.get_args(frame))
+        probe = cast(LogFunctionProbe, self.probe)
+        _args = list(self.args or safety.get_args(frame))
 
         if probe.evaluate_at == ProbeEvaluateTimingForMethod.EXIT:
             return
 
-        if not self._evalCondition(dict(_args)):
+        if not self._eval_condition(dict(_args)):
             return
 
-        if probe.limiter.limit() is RateLimitExceeded:
-            self.state = CaptureState.SKIP_RATE
-            return
+        if probe.capture_snapshot:
+            if probe.limiter.limit() is RateLimitExceeded:
+                self.state = CaptureState.SKIP_RATE
+                return
+
+            self.entry_capture = _captured_context(
+                _args,
+                [],
+                (None, None, None),
+                limits=probe.limits,
+            )
 
-        self.entry_capture = _captured_context(
-            _args,
-            [],
-            (None, None, None),
-            limits=probe.capture,
-        )
+        if probe.evaluate_at == ProbeEvaluateTimingForMethod.ENTER:
+            self._eval_message(dict(_args))
+            self.state = CaptureState.DONE_AND_COMMIT
 
     def exit(self, retval, exc_info, duration):
-        probe = cast(SnapshotFunctionProbe, self.probe)
+        probe = cast(LogFunctionProbe, self.probe)
         _args = self._enrich_args(retval, exc_info, duration)
 
         if probe.evaluate_at == ProbeEvaluateTimingForMethod.EXIT:
-            if not self._evalCondition(_args):
-                return
-            if probe.limiter.limit() is RateLimitExceeded:
-                self.state = CaptureState.SKIP_RATE
+            if not self._eval_condition(_args):
                 return
+            if probe.capture_snapshot:
+                if probe.limiter.limit() is RateLimitExceeded:
+                    self.state = CaptureState.SKIP_RATE
+                    return
         elif self.state != CaptureState.NONE:
             return
 
         _locals = []
         if exc_info[1] is None:
             _locals.append(("@return", retval))
 
-        self.return_capture = _captured_context(
-            self.args or safe_getter.get_args(self.frame), _locals, exc_info, limits=probe.capture
-        )
+        if probe.capture_snapshot:
+            self.return_capture = _captured_context(
+                self.args or safety.get_args(self.frame), _locals, exc_info, limits=probe.limits
+            )
         self.duration = duration
         self.state = CaptureState.DONE_AND_COMMIT
+        if probe.evaluate_at != ProbeEvaluateTimingForMethod.ENTER:
+            self._eval_message(dict(_args))
 
     def line(self, _locals=None, exc_info=(None, None, None)):
         frame = self.frame
-        probe = cast(SnapshotLineProbe, self.probe)
+        probe = cast(LogLineProbe, self.probe)
```

Comment:
I think we should turn this `cast` into an `isinstance` check in this case.

Reply:
refactored